File: package/main/src/Object/invert.ts
/**
 * Creates a new object with keys and values swapped.
 * If multiple keys share the same value, later entries overwrite earlier ones.
 *
 * @template K - Key type of the input object (must be a PropertyKey)
 * @template V - Value type of the input object (must be a PropertyKey)
 * @param {Record<K, V>} object - Source object
 * @returns {Record<V, K>} A new object with keys and values swapped
 * @example
 * invert({ a: 1, b: 2 }); // { 1: "a", 2: "b" }
 */
export const invert = <K extends PropertyKey, V extends PropertyKey>(
  object: Record<K, V>,
): Record<V, K> => {
  const result = {} as Record<V, K>;
  for (const key of Object.keys(object) as K[]) {
    const value = object[key];
    result[value] = key;
  }
  return result;
};

File: package/umt_python/src/object/invert.py
from typing import TypeVar

K = TypeVar("K")
V = TypeVar("V")


def invert(object_: dict[K, V]) -> dict[V, K]:
    """
    Creates a new object with keys and values swapped.
    If multiple keys share the same value, later entries overwrite earlier ones.

    The input object is not mutated.

    Args:
        object_ (dict[K, V]): Source object.

    Returns:
        dict[V, K]: A new object with keys and values swapped.

    Example:
        >>> invert({"a": 1, "b": 2})
        {1: 'a', 2: 'b'}
        >>> invert({"x": "X", "y": "Y"})
        {'X': 'x', 'Y': 'y'}
        >>> invert({"a": 1, "b": 1})
        {1: 'b'}
    """
    return {value: key for key, value in object_.items()}
